tests/src/whisk/core/controller/test/EntitlementProviderTests.scala

```diff
@@ -180,4 +184,240 @@ class EntitlementProviderTests extends ControllerTestCommon {
         Await.result(entitlementProvider.check(adminUser, DELETE, one), requestTimeout) should not be (true)
         Await.result(entitlementProvider.revoke(adminUser.subject, READ, one), requestTimeout) // revoked
     }
+
+    behavior of "Package Collection"
+
+    it should "only allow read access for listing package collection" in {
+        implicit val tid = transid()
+        implicit val ep = entitlementProvider
+
+        val paths = Seq(
+            (READ, someUser, Right(true)),
+            (PUT, someUser, Right(false)),
+            (DELETE, someUser, Right(false)),
+            (ACTIVATE, someUser, Right(false)),
+            (REJECT, someUser, Right(false)),
+
+            (READ, guestUser, Right(true)),
+            (PUT, guestUser, Right(false)),
+            (DELETE, guestUser, Right(false)),
+            (ACTIVATE, guestUser, Right(false)),
+            (REJECT, guestUser, Right(false)))
+
+        paths.map {
+            case (priv, who, expected) =>
+                val check = new PackageCollection(entityStore).implicitRights(
+                    who,
+                    Set(who.namespace()),
+                    priv,
+                    // any user can list any namespace packages
+                    // (because this performs a db view lookup which is later filtered)
+                    Resource(someUser.namespace.toPath, PACKAGES, None))
+                Await.ready(check, requestTimeout).eitherValue.get shouldBe expected
```

Comment:
this whole suite runs in 3 seconds - eh...
I do intend to replace the couchdb backing store with the mock store coming up in a new pr.